src/conversions/std/time.rs

```diff
@@ -0,0 +1,153 @@
+
+        let delta = Duration::from_secs(DAY_SECONDS);
+        check("delta 1 day", delta, 1, 0, 0);
+
+        let delta = Duration::new(7 * DAY_SECONDS + 1, 999999000);
+        check("delta 7 day and a seconds", delta, 7, 1, 999999);
+
+        // Check the maximum value allowed by PyDelta, which is different
+        // from the maximum value allowed in Duration. This should pass.
+        let delta = Duration::new(86399999999999, 999999000);
+        check("delta max value", delta, 999999999, 86399, 999999);
+
+        // Check that trying to convert an out of bound value panics.
+        Python::with_gil(|py| {
+            assert!(panic::catch_unwind(|| Duration::MAX.to_object(py)).is_err());
+        });
+    }
+
+    #[test]
+    fn test_std_duration_frompyobject() {
+        // Utility function used to check different durations.
+        // The `name` parameter is used to identify the check in case of a failure.
+        let check = |name: &'static str, delta: Duration, py_days, py_seconds, py_ms| {
+            Python::with_gil(|py| {
+                let py_delta = PyDelta::new(py, py_days, py_seconds, py_ms, true).unwrap();
+                let py_delta: Duration = py_delta.extract().unwrap();
+                assert_eq!(py_delta, delta, "{}: {:?} != {:?}", name, py_delta, delta);
+            })
+        };
+
+        check("delta zero", Duration::ZERO, 0, 0, 0);
+
+        let delta = Duration::from_secs(DAY_SECONDS);
+        check("delta 1 day", delta, 1, 0, 0);
+
+        let delta = Duration::new(7 * DAY_SECONDS + 1, 999999000);
+        check("delta 7 day and a seconds", delta, 7, 1, 999999);
+
+        // Check the maximum value allowed by PyDelta, which is different
+        // from the maximum value allowed in Duration. This should pass.
+        check(
+            "max pydelta value",
+            Duration::new(86399999999999, 999999000),
+            999999999,
+            86399,
+            999999,
+        );
+
+        // This check is to assert that we can't construct every possible Duration from a PyDelta
+        // since they have different bounds.
+        Python::with_gil(|py| {
+            let low_days: i32 = -1000000000;
+            // This panics on PyDelta::new
+            assert!(panic::catch_unwind(|| {
+                let pydelta = PyDelta::new(py, low_days, 0, 0, true).unwrap();
+                if let Ok(_duration) = pydelta.extract::<Duration>() {
+                    // So we should never get here
+                }
+            })
+            .is_err());
+
+            let high_days: i32 = 1000000000;
+            // This panics on PyDelta::new
+            assert!(panic::catch_unwind(|| {
+                let pydelta = PyDelta::new(py, high_days, 0, 0, true).unwrap();
+                if let Ok(_duration) = pydelta.extract::<Duration>() {
+                    // So we should never get here
+                }
+            })
+            .is_err());
+        });
```

Comment:
Maybe rather than having the `.unwrap()` on the `PyDelta` construction, it's enough to have something like:

```rust
assert!(PyDelta::new(py, low_days, 0, 0, true).is_err());
let _ = Duration::new(low_days * DAY_SECONDS, 0); // doesn't panic
```

However, it looks like Python durations can be negative, whereas I think `std::time::Duration` cannot hold negative values, so these tests may want to check that behaviour.